dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java

```diff
@@ -208,7 +212,11 @@ public AnalyticsTableType getAnalyticsTableType()
 
         List<AnalyticsTable> tables = new ArrayList<>();
 
-        List<Program> programs = idObjectManager.getAllNoAcl( Program.class );
+        List<Program> programs = params.getSkipPrograms().size() == 0 ? idObjectManager.getAllNoAcl( Program.class )
+            : idObjectManager.getAllNoAcl( Program.class )
+                .stream()
```

Comment:
Note that we want to use the *UID* of the programs to identify them here, not the name. The UID is the main ID.

Reply:
Done